FinniversKit/Sources/Components/BrazePromotionView/BrazePromotionView.swift

```diff
@@ -0,0 +1,249 @@
+        verticalStackView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor, constant: -.spacingM),
+        verticalStackView.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor, constant: -.spacingM),
+    ]
+
+    private var viewModel: BrazePromotionViewModel
+    private var imageDatasource: RemoteImageViewDataSource?
+
+    // MARK: - Public properties
+
+    public enum Action {
+        case primary
+        case secondary
+    }
+
+    public weak var delegate: BrazePromotionViewDelegate?
+
+    // MARK: - Init
+
+    public init(viewModel: BrazePromotionViewModel, imageDatasource: RemoteImageViewDataSource) {
+        self.viewModel = viewModel
+        self.imageDatasource = imageDatasource
+        super.init(frame: .zero)
+        setup()
+        configure(with: viewModel, imageDatasource: imageDatasource)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(with viewModel: BrazePromotionViewModel, imageDatasource datasource: RemoteImageViewDataSource) {
+        titleLabel.text = viewModel.title
+        primaryButton.configure(withTitle: viewModel.primaryButtonTitle)
+
+        if let text = viewModel.text {
+            textLabel.text = text
+        } else {
+            textLabel.isHidden = true
+        }
+
+        loadImage()
+        if let dismissible = viewModel.dismissible, !dismissible {
+            closeButton.removeFromSuperview()
+        } else {
+            backgroundView.bringSubviewToFront(closeButton)
+        }
+    }
+
+    private func loadImage() {
+        guard let imageUrl = viewModel.image else {
+            remoteImageView.removeFromSuperview()
+            NSLayoutConstraint.deactivate(stackViewConstraintsImage)
+            NSLayoutConstraint.activate(stackViewConstraintsNoImage)
+            closeButton.setImage(UIImage(named: .cross).withTintColor(.textPrimary), for: .normal)
+            closeButton.backgroundColor = UIColor.clear
+            return
+        }
+        NSLayoutConstraint.deactivate(stackViewConstraintsNoImage)
+        NSLayoutConstraint.activate(stackViewConstraintsImage)
+        backgroundView.addSubview(remoteImageView)
+        remoteImageView.dataSource = imageDatasource
+        NSLayoutConstraint.activate([
+            remoteImageView.widthAnchor.constraint(equalTo: backgroundView.widthAnchor, multiplier: 0.3),
+            remoteImageView.topAnchor.constraint(equalTo: backgroundView.topAnchor),
+            remoteImageView.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor),
+            remoteImageView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor)
+        ])
+        remoteImageView.loadImage(
+            for: imageUrl,
+            imageWidth: 100,
+            fallbackImage: UIImage(named: .noImage)
+        )
+    }
```

Comment:
And, depending on whether or not the view should be reconfigurable, I think this reads a bit better if it doesn't need to be that :)

```swift
private func loadImage() {
    if let imageUrl = viewModel.image {
        closeButton.setImage(UIImage(named: .cross), for: .normal)
        closeButton.backgroundColor = UIColor.black.withAlphaComponent(0.6)

        backgroundView.addSubview(remoteImageView)
        remoteImageView.dataSource = imageDatasource

        NSLayoutConstraint.activate(stackViewConstraintsImage)
        
        NSLayoutConstraint.activate([
            remoteImageView.widthAnchor.constraint(equalTo: backgroundView.widthAnchor, multiplier: 0.3),
            remoteImageView.topAnchor.constraint(equalTo: backgroundView.topAnchor),
            remoteImageView.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor),
            remoteImageView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor)
        ])

        remoteImageView.loadImage(
            for: imageUrl,
            imageWidth: 100,
            fallbackImage: UIImage(named: .noImage)
        )
    } else {
        closeButton.setImage(UIImage(named: .cross).withTintColor(.textPrimary), for: .normal)
        closeButton.backgroundColor = UIColor.clear
        
        NSLayoutConstraint.activate(stackViewConstraintsNoImage)
    }
}
```